conn.go

```diff
@@ -771,10 +773,26 @@ func (c *conn) writeProtoHeader(pID protoID) error {
 	if c.connectTimeout != 0 {
 		_ = c.net.SetWriteDeadline(time.Now().Add(c.connectTimeout))
 	}
-	_, err := c.net.Write([]byte{'A', 'M', 'Q', 'P', byte(pID), 1, 0, 0})
+	_, err := c.netWriteWithRetry([]byte{'A', 'M', 'Q', 'P', byte(pID), 1, 0, 0})
 	return err
 }
 
+func (c *conn) netWriteWithRetry(b []byte) (int, error) {
+	var n int
+	var err error
+	for i := 0; i < 5; i++ {
```

Comment:
Thinking about this - if I were trying to close the connection would I be blocked waiting for this net.Write() to complete? Ie, how do I cancel out of this if we're trying to cleanup and get rid of the connection?

Reply:
You wouldn't be blocked, but you would be delayed until either the write succeeds or the retry count is exceeded.

Reply:
Will add a check to see if the mux has been closed to exit early.